crates/agglayer-types/src/lib.rs

```diff
@@ -187,10 +187,34 @@ impl std::fmt::Display for GenerationType {
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub enum CertificateStatus {
+    /// Received certificate from the network, nothing checked yet.
+    ///
+    /// Certificate will stay in this state until the L1InfoRoot is finalized.
+    /// It will then be queued to be sent to agglayer-prover for PP generation.
     Pending,
+
+    /// Pessimistic proof has been generated for the certificate and stored in
+    /// the rocksdb in the agglayer node.
     Proven,
+
+    /// Certificate proof has been acknowledged by L1, but the contract call has
+    /// not returned its verification feedback yet.
+    ///
+    /// Waiting for L1 to execute the contract call.
```

Comment:
```suggestion
    /// Settlement of the certificate's proof has been started on L1 by issuing a contract call to the rollupmanager,
    /// the associated transaction has not yet enough confirmations.
    ///
    /// The certificate can move from this status to Settled if the associated transaction is accepted
    /// and when the transaction receipt is a success. In case of the transaction receipt being in failure,
    /// the certificate end up in Error
```